mlir/patches/use-ir-from-jax.patch

```diff
@@ -0,0 +1,10 @@
+--- build/python_packages/quantum/mlir_quantum/ir.py	2023-03-07 15:06:48.523539871 -0500
```

Comment:
One question for my understanding:
- Why is the path here specified starting at `catalyst/mlir`, but when we apply the patch we first move to `catalyst/mlir/build/python_packages/quantum/mlir_quantum`?

Reply:
Generating the other patch I suggested I now realize this is because you ran the diff from the `catalyst/mlir` folder.
Oddly enough the `patch` still will only look for `ir.py` in the current directory rather than the full path specified here.

Reply:
I just meant it's curious that running `patch < ${PATCH}` from the directory that is the root of the file path specified in the patch itself doesn't work.
That is, given the patch specifies `build/python_packages/quantum/mlir_quantum/ir.py` as the file path, which is located in `catalyst/mlir/`, you would think that you could run the `patch` command from `catalyst/mlir/` (without needing the `-d` option).

Anyway, it's not too important.

Reply:
It is not an unnecessary detail. The "build" directory is different in the patch file and in the CI/CD.

Reply:
True but I tried it locally with the same path structure and was not able to find the file referenced in the patch 😅 

EDIT: ok it works if you set the option `-p 0` ("strip number of leading components from the file path). I wonder what the default there is 🤔